db-scripts/src/main/resources/migration.sql

```diff
@@ -380,4 +380,49 @@ ALTER TABLE `mutation` ADD COLUMN `DRIVER_FILTER_ANNOTATION` VARCHAR(80) NULL;
 ALTER TABLE `mutation` ADD COLUMN `DRIVER_TIERS_FILTER` VARCHAR(50) NULL;
 ALTER TABLE `mutation` ADD COLUMN `DRIVER_TIERS_FILTER_ANNOTATION` VARCHAR(80) NULL;
 
-UPDATE info SET DB_SCHEMA_VERSION="2.4.0";
+##version: 2.4.1
+-- ========================== new reference genome genes related tables =============================================
+CREATE TABLE `reference_genome` (
```

Comment:
The field definitions here should be updated to NOT NULL.

Reply:
@n1zea144 , I've solved the mapping issue with those gene having multiple genetic loci. Again, rebased to the latest rc, tested, and pushed the changes back to the origin. Thanks!

Reply:
Getting NULL and "-" also returned in substring_index, need to determine why.

![image](https://user-images.githubusercontent.com/366003/32461598-659b527e-c304-11e7-979a-03d8c5dfb288.png)

Reply:
@n1zea144 , can you please post ENTREZ_GENE_ID, HUGO_GENE_SYMBOL for those NULLs and "-" s?

Reply:
@n1zea144 , I do not think NULL and '-' were introduced by the query, just would like to double check if "cytoband" fields were nulls or contained '-'.

Reply:
Hi @khzhu  I think there are some records with '-' as cytoband, but I wonder if query does cause NULL.  I sent some info in slack.

Reply:
hi @n1zea144 , from my testing those NULL values came from null "cytoband" fields originally in the gene table. There are a total of 60072 rows in the gene table (we populated the gene table from the seed database downloaded from the cBioPortal datahub) of our production database and 20414 rows have null cytoband. (60072-20414)=39658 which is exactly number returned by the substring_index.

Reply:
@khzhu I placed a dump of the gene table I've been using here: [cBioPortal production gene-table](http://cbio.mskcc.org/~grossb/cbioportal-mysql/cgds_gdac_gene.sql.gz) to speed up testing.  We know some cytoband contains "-" only, I think the question to answer is what are all the conditions in which your substring returns "NULL".

Reply:
@n1zea144 , sorry I was at other building till now. thanks, that is really helpful, I am going to restore your gene table and do more testing.

Reply:
Hi @n1zea144 , I've done my testing on your gene table and confirmed that substring_index call returned NULL when cytoband == NULL (please see attached screenshot). And, the PR has been updated with the latest changes. Thanks again for the help!! 
<img width="1119" alt="screen shot 2017-11-07 at 5 39 16 pm" src="https://user-images.githubusercontent.com/17375786/32522046-b6dc85ce-c3e3-11e7-9dc8-8461cb76258c.png">

Reply:
@khzhu Ok, with latest update, an exception is thrown trying to backfill reference_genome_gene since some values (as we've already discovered are null) are NULL.  I think we have to relax constraint on table definition to allow for nulls in cytoband, or maybe better - set NULL values to NA (not available?).

Reply:
@n1zea144 , I would vote for "relax constraint on table definition to allow for nulls".

Reply:
hi @n1zea144, pushed the change back to the origin, thanks!